### PR TITLE
Clear Site cache upon SiteConfiguration change.

### PR DESCRIPTION
**Pull request by mjfrey**
Call to Site clear_cache when SiteConfiguration changes.

### CODE REVIEW

**Comment by mjfrey** [comment]
@clintonb @douglashall

File: ecommerce/core/models.py [hunk 1]
```diff
@@ -187,3 +190,11 @@ def validate_configuration():
     """ Validates all existing SiteConfiguration models """
     for config in SiteConfiguration.objects.all():
         config.clean_fields()
+
+
+@receiver(post_save, sender=SiteConfiguration)
```
Comment (clintonb) [inline ecommerce/core/models.py]: You can avoid the signal work and simply override `SiteConfiguration.save()`.

**Comment by mjfrey** [comment]
@douglashall @clintonb take a second look please.

File: ecommerce/core/models.py [hunk 2]
```diff
@@ -147,6 +148,13 @@ def clean_fields(self, exclude=None):
     def segment_client(self):
         return SegmentClient(self.segment_key, debug=settings.DEBUG)
 
+    def save(self, *args, **kwargs):
+        """
+        Clear Site cache upon SiteConfiguration changed
```
Comment (clintonb) [inline ecommerce/core/models.py]: This should simply be a comment above the line, rather than a docstring.

**Comment by clintonb** [comment]
👍

**Comment by douglashall** [comment]
@mjfrey @clintonb Just realized that this will not work in a multi-process environment because Django is caching sites in memory.

**Comment by clintonb** [comment]
https://code.djangoproject.com/ticket/15894

We should not be frequently updating SiteConfiguration. The solution here may be to simply restart the service when changes are made, or find a way to expose a VPN-protected endpoint that can be used to trigger the cache clear on all processes. It's not ideal; but, it works.

**Pull request merged**